c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -589,7 +700,28 @@ class DeviceCachingAllocator {
     return subsumed_size;
   }
 
-  BlockPool& get_pool(size_t size) {
+  BlockPool& get_pool(size_t size, cudaStream_t stream) {
+    // captures_underway is a conservative guess that the current stream may be capturing.
+    // It's only > 0 if some thread has begun and not yet ended a capture, so it's usually 0,
+    // and we can short-circuit cudaStreamCaptureStatus (which does a TLS lookup).
+    if C10_UNLIKELY(captures_underway) {
```

Comment:
nit: missing enclosing parens (yes the macro saves you but it still looks weird)